arangod/Aql/WindowNode.cpp

```diff
@@ -291,7 +291,8 @@ void WindowBounds::toVelocyPack(VPackBuilder& b) const {
               append(duration.seconds, 'S');
             } else {
               result.append(std::to_string(duration.seconds)).push_back('.');
-              append(duration.milliseconds, 'S');
+              auto ms = std::to_string(duration.milliseconds);
+              result.append(3 - ms.size(), '0').append(ms).push_back('S');
```

Comment:
Not sure if this is safe in all cases, e.g. if milliseconds has a value of 1000 or more.
Maybe this is not possible because milliseconds are bounded elsewhere. But it is not immediately clear here if that is the case. So I suggest we limit `ms` to a length of 3 characters here to be on the safe side.

Reply:
Yes, the number of decimals is limited to 3 in `parseIsoDuration`, but I can at least add an assertion.